tests/webTables.spec.ts

```diff
@@ -0,0 +1,295 @@
+    await page.getByText("Veterinarians").click()
+    await page.getByText("All").click()
+
+    /* 10. On the Veterinarians page, add assertion that "Rafael Ortega" has specialty "dermatology" */
+    // Validate Raphael's veterinarian specialty
+    await expect(page.getByRole('row', {name: 'Rafael Ortega'}).getByRole('cell').nth(1)).toHaveText('dermatology')
+
+    /* 11. Navigate to SPECIALTIES page, revert the changes renaming "dermatology" back to "surgery" */
+    // Navigate to the Specialties page
+    await page.getByRole('link', {name: 'Specialties'}).click()
+
+   // Iterate through rows to find 'dermatology'
+   for (let row of allRows) {
+        // If 'surgery' is found...
+        if( await row.locator('input').inputValue() === 'dermatology'){
+            // ... click 'Edit'
+            await row.getByRole('button', { name: 'Edit' }).click()
+            break;
+        }
+    }
+
+    // Await 'surgery' to load in the input...
+    await expect(specialtyInput).toHaveValue('dermatology')
+
+    // Revert dermatology back to surgery
+    await specialtyInput.clear()
+    await specialtyInput.fill('surgery')
+    await page.getByText('Update').click()
+
+})
+
+
+test('Validate specialty lists', async ({ page }) => {
+   
+    /* 1. Select the SPECIALTIES menu item in the navigation bar */ 
+    // Navigate to the Specialties page
+    await page.getByRole('link', {name: 'Specialties'}).click()
+
+    await page.waitForSelector('tbody')
+    /* 2. On the Specialties page, select "Add" button. Type the new specialty "oncology" and click "Save" button */
+    // Click on 'Add' button to enter a new specialty, oncology
+    await page.getByRole('button', {name: 'Add'}).click()
+    await page.locator('#name').fill('oncology')
+    await page.getByText('Save').click()
+    
+    // Validate 'oncology' was added
+    await expect(page.locator('tbody tr').last().getByRole('textbox')).toHaveValue('oncology')
+
+     /* 3. Extract all values of specialties and put them into the array. */
+    // Collect all specialties and store for later usage
+    const allSpecialtyInputs = await page.locator('tbody tr').getByRole('textbox').all()
+    const allSpecialties : string [] = []
+
+    // Iterate through all the rows to collect the Specialty and store in allSpecialties for future validation
+    for(const input of allSpecialtyInputs){
+        allSpecialties.push(await input.inputValue())
+    }
+
+    /* 4. Select the VETERINARIANS menu item in the navigation bar, then select "All" */
+    // Navigate to the 'Veterinarians' page through the dropdown menu at the top
+    await page.getByText("Veterinarians").click();
+    await page.getByText("All").click()
+    
+    /* 5. On the Veterinarians page, locate the "Sharon Jenkins" in the list and click "Edit" button */
+    // Locate Sharon Jenkins and click 'Edit'
+    await page.getByRole('row', {name: 'Sharon Jenkins'}).getByRole('cell').last().getByRole('button', {name: 'Edit Vet'}).click()
```

Comment:
This can be simplified:
`await page.getByRole('row', {name: 'Sharon Jenkins'}).getByRole('button', {name: 'Edit Vet'}).click()`

So you find the row with a desired name, and then within that row find the button. You don't need to target the cell where button is located. Playwright will do it for you.

Reply:
Oops, looks like I did!  I cleaned these up and found one other thing to fix.  Thanks for the review!